packages/playwright/src/worker/testInfo.ts

```diff
@@ -193,7 +195,7 @@ export class TestInfoImpl implements TestInfo {
     this._attachmentsPush = this.attachments.push.bind(this.attachments);
     this.attachments.push = (...attachments: TestInfo['attachments']) => {
       for (const a of attachments)
-        this._attach(a.name, a);
+        this._attach(a, this._expectStepId() ?? this._parentStep()?.stepId);
```

Comment:
Instead of reaching for the `_expectStepId()` here, I'd prefer the expect matchers to attach files by passing an explicit step, or returning it the step complete callback. Ideally, the dependencies flow expect -> testInfo, not vice versa.

Reply:
After thinking a bit more about this, the current approach seems good. Please ignore this comment.